File: C#/GCD/GCD/ExtEuclid.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Numerics;
using System.Text;
using System.Threading.Tasks;

namespace GCD
{
    class ExtEuclid : GCD
    {
        public ExtEuclid()
        {
            this.Name = "Extended Euclid";
        }
        
        public override BigInteger[] compute(BigInteger p, BigInteger q, bool debug = false)
        {
            BigInteger oldR = p, newR = q,
                       oldS = 1, newS = 0,
                       oldT = 0, newT = 1;

            while (newR != 0)
            {
                BigInteger quotient = oldR / newR;

                BigInteger temp = newR;
                newR = oldR - quotient * temp;
                oldR = temp;

                temp = newS;
                newS = oldS - quotient * temp;
                oldS = temp;

                temp = newT;
                newT = oldT - quotient * temp;
                oldT = temp;
            }

            // Bezout coefficients oldS and oldT
            return new BigInteger[] { oldR, oldS, oldT } ;
        }
    }
}

File: python/ExtEuclid.py
from GCD import GCD

class ExtEuclid(GCD):
    def __init__(self):
        super(ExtEuclid, self).__init__("Extended Euclid")
    
    def compute(self, p, q, debug = False):
        # Used https://en.wikipedia.org/wiki/Extended_Euclidean_algorithm as reference

        old_r = p
        new_r = q

        old_s = 1
        new_s = 0

        old_t = 0
        new_t = 1
        count = 0
        
        while (new_r is not 0):
            quotient = old_r // new_r

            old_r, new_r = new_r, old_r - quotient * new_r
            old_s, new_s = new_s, old_s - quotient * new_s
            old_t, new_t = new_t, old_t - quotient * new_t

            count += 1

        return (old_r, old_s, old_t, count)
        
